File: LeetCode/CodingInterviews/fibonacci.go
package coding

func fib(n int) int {
	if n < 2 {
		return n
	}
	var a, b = 0, 1
	var res int
	for i := 2; i <= n; i++ {
		res = (a + b) % 1000000007
		a = b
		b = res
	}
	return res
}

File: LeetCode/CodingInterviews/Java/Fibonacci.java
package LeetCode;

// leetcode 509
class Solution {
    public int fib(int n) {
        if (n <= 1)
            return n;
        int ans = 1, a = 0, b = 1;
        for (int i = 2; i <= n; i++) {
            ans = (a + b) % ((int) 1e9 + 7);
            a = b;
            b = ans;
        }
        return ans;
    }
}